File: 백준/Silver/2193. 이친수/이친수.py
import sys
import os

# 운영 체제 확인 및 입력 파일 설정 (필요한 경우)
if os.name == 'nt':  # Windows 환경
    sys.stdin = open('input.txt', 'r')

n = int(input())

dp = [0, 1]

for i in range(2, n + 1):
    dp.append(dp[i - 2] + dp[i - 1])

print(dp[n])

File: 백준/Silver/2193. 이친수/이친수.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";

let input = fs.readFileSync(filePath)
    .toString()
    .trim()
// .split('\n')
// .map(x => x.trim());


function solution(n) {
    n = Number(n);
    let dp = [0, 1];
    for (let i = 2; i <= n; i++) {
        dp[i] = BigInt(dp[i - 2]) + BigInt(dp[i - 1]);
    }
    console.log(dp[n].toString());
}

solution(input);
